services/certificate/certificate-expiration.service.js

```diff
@@ -0,0 +1,110 @@
+import emojic from 'emojic'
+import Joi from 'joi'
+import { BaseService, pathParam, queryParam } from '../index.js'
+import trace from '../../core/base-service/trace.js'
+
+const queryParamSchema = Joi.object({
+  warningDays: Joi.number().min(0),
+  dangerDays: Joi.number().min(0),
+}).required()
+
+export default class CertificateExpiration extends BaseService {
+  static category = 'monitoring'
+
+  static route = {
+    base: 'certificate/expiration',
+    pattern: ':hostname',
+    queryParamSchema,
+  }
+
+  static openApi = {
+    '/certificate/expiration/{hostname}': {
+      get: {
+        summary: 'Certificate Expiration',
+        parameters: [
+          pathParam({ name: 'hostname', example: 'shields.io' }),
+          queryParam({
+            name: 'warningDays',
+            schema: { type: 'number' },
+            description:
+              'Warning (yellow) if less than this many days remain (default=30)',
+            example: '30',
+          }),
+          queryParam({
+            name: 'dangerDays',
+            schema: { type: 'number' },
+            description:
+              'Danger (red) if less than this many days remain (default=7)',
+            example: '7',
+          }),
+        ],
+      },
+    },
+  }
+
+  static _cacheLength = 0
```

Comment:
This is the length of time badges will be cached at the CDN. I didn't spot this first time round, but this can't be zero.

Any strong reason this should be different from the default for the monitoring category (2 mins)?

Reply:
removed